Meshtastic/Helpers/MeshPackets.swift

```diff
@@ -808,7 +808,12 @@ func textMessageAppPacket(
 			let fetchedUsers = try context.fetch(messageUsers)
 			let newMessage = MessageEntity(context: context)
 			newMessage.messageId = Int64(packet.id)
-			newMessage.messageTimestamp = Int32(bitPattern: packet.rxTime)
+			/// For message display if the rx time is off by more than 20 seconds set the timestamp to now to assist sorting
+			if Date().timeIntervalSince1970 < Double(packet.rxTime - 10000) || Date().timeIntervalSince1970 > Double(packet.rxTime + 10000) {
```

Comment:
Logic looks sound. I guess the only caveat is that this will potentially always be true for messages that are queued up in the phone API on a disconnected device